app/ssttpcalculator/CalculatorController.scala

```diff
@@ -53,19 +53,19 @@ class CalculatorController @Inject() (
 
   def getTaxLiabilities: Action[AnyContent] = as.authorisedSaUser.async { implicit request =>
     JourneyLogger.info(s"CalculatorController.getTaxLiabilities: $request")
-    journeyService.getEligibleJourneyInProgress().map { journey =>
+    journeyService.authorizedForSsttp { journey: Journey =>
```

Comment:
hmmm... So the user is already authorised, what happened in an action call `as.authorisedSaUse`. At this point users is already eligible thus we called `journeyService.getEligibleJourneyInProgress()...` . 
Why do we actually need to call the second time ` journeyService.authorizedForSsttp ` and is there any reason why not eligible users access this page?

Reply:
`authorisedSaUser` authorises against credentials while `authorizedForSsttp` checks the Journey `eligibilityStatus` so it's probably misnamed.

`getEligibleJourneyInProgress` and `authorizedForSsttp` have largely similar behaviour:

```
def getEligibleJourneyInProgress()(implicit request: Request[_]): Future[Journey] = Mdc.preservingMdc {
    getJourney().map {
      case j: Journey if j.status == InProgress && j.maybeEligibilityStatus.isDefined && j.eligibilityStatus.eligible => j
      case j => throw new RuntimeException(s"Expected eligible journey in progress [${j}]")
    }
  }
```

```
  def authorizedForSsttp(block: Journey => Future[Result])(implicit request: Request[_]): Future[Result] = {
    JourneyLogger.info(s"${this.getClass.getSimpleName}: $request")

    for {
      journey <- getJourney()
      result <- journey match {
        case journey if journey.status == InProgress =>
          JourneyLogger.info(s"${this.getClass.getSimpleName}.authorizedForSsttp: currentSubmission", journey)
          journey.requireIsEligible()
          block(journey)

        case journey if journey.status == FinishedApplicationSuccessful =>
          JourneyLogger.info(s"${this.getClass.getSimpleName}.authorizedForSsttp: currentSubmission", journey)
          Future.successful(Results.Redirect(ssttparrangement.routes.ArrangementController.applicationComplete()))

      }
    } yield result
```

The latter checks eligibilityStatus by asserting over journey state through `journey.requireIsEligible` while `getEligibleJourneyInProgress` fails with an exception when encountering a complete journey (and consequently technical difficulties)

As you said, this is a band-aid solution since the underlying problem is a UX one but this would
1) reduce the number of technical difficulties screens the user sees
2) reduce the number of noise in Kibana so actual errors are more visible